### PR TITLE
Basic refactoring to parameterise..

### PR DESCRIPTION
**Pull request by adusumillipraveen**
Did some basic refactoring to parameterise  properties

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/filter/webapp/SpringSecurityAuthenticationProvider.java [hunk 1]
```diff
@@ -1,211 +1,206 @@
...
-        AuthorizationService authorizationService = engine.getAuthorizationService();
-        Authorization authorization = authorizationService.createNewAuthorization(Authorization.AUTH_TYPE_GRANT);
+    AuthorizationService authorizationService = engine.getAuthorizationService();
+    Authorization authorization = authorizationService
+        .createNewAuthorization(Authorization.AUTH_TYPE_GRANT);
 
-        boolean cmcAdminAuthExists = authorizationService.createAuthorizationQuery().resourceType(Resources.APPLICATION)
-                .groupIdIn(CMC_ADMIN_CAMUNDA_GROUP).resourceId("*").count() > 0;
-        if (!cmcAdminAuthExists) {
+//        boolean cmcAdminAuthExists = authorizationService.createAuthorizationQuery().resourceType(Resources.APPLICATION)
```
Comment (adusumillipraveen) [inline src/main/java/uk/gov/hmcts/reform/camunda/bpm/filter/webapp/SpringSecurityAuthenticationProvider.java]: Can we delete this ?

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/config/GroupConfig.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.reform.camunda.bpm.config;
+
+public class GroupConfig {
+
+  private String adGroupId;
+
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/camunda/bpm/config/GroupConfig.java]: prob don't need all these spaces between fields? other classes don't have

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/context/SpringContext.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.reform.camunda.bpm.context;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.stereotype.Component;
+@Component
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/camunda/bpm/context/SpringContext.java]:
```suggestion

@Component
```

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/context/SpringContext.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+public class SpringContext implements ApplicationContextAware {
+  @Autowired
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/camunda/bpm/context/SpringContext.java]:
```suggestion
  
  @Autowired
```

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/context/SpringContext.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.reform.camunda.bpm.context;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.stereotype.Component;
+@Component
+public class SpringContext implements ApplicationContextAware {
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/camunda/bpm/context/SpringContext.java]: Maybe a javadoc comment on why this class is needed?

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/filter/webapp/SpringSecurityAuthenticationProvider.java (hunk 1 above)
Comment (timja) [reply src/main/java/uk/gov/hmcts/reform/camunda/bpm/filter/webapp/SpringSecurityAuthenticationProvider.java]: yeah

File: src/main/java/uk/gov/hmcts/reform/camunda/bpm/filter/webapp/SpringSecurityAuthenticationProvider.java [hunk 6]
```diff
@@ -1,211 +1,206 @@
...
+                .count() == 0) {
+              identityService
+                  .createTenantGroupMembership(groupConfig.getTenantId(), groupConfig.getGroupId());
             }
-
-//            if (identityService.createUserQuery().memberOfGroup(CMC_ADMIN_CAMUNDA_GROUP).count() == 0) {
-                identityService.createMembership(id, CMC_ADMIN_CAMUNDA_GROUP);
-//            }
-            camundaGroups.add(CMC_ADMIN_CAMUNDA_GROUP);
+            //  if (identityService.createUserQuery().memberOfGroup(groupConfig.getGroupId()).count() == 0) {
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/camunda/bpm/filter/webapp/SpringSecurityAuthenticationProvider.java]: delete?

File: src/main/resources/application.yaml [hunk 7]
```diff
@@ -41,3 +43,35 @@ server:
...
     context-path: /
   port: 8999
+camundaGroups:
+  cmcadmin:
+    adGroupId: d6eb4b7b-d156-4cc4-918c-5de9d8e7ad5b
+    tenantName: cmc
+    tenantId: Civil Money Claims
+    groupId: cmc-admin
+    groupName: CMC Admin
+    accessControl:
```
Comment (timja) [inline src/main/resources/application.yaml]: Not sure if we need this fine grain control here, should probably be done by a type of group not specifying everything here.
(note: goal is to be able to externalise the config through environment variables / whatever, so the less information required the better)

**Pull request merged**